.github/workflows/test_plugin_uploader.yml

```
@@ -0,0 +1,68 @@
+name: Test Plugin Uploader
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'resources/plugin_uploader/**'
+      - '.github/workflows/test_plugin_uploader.yml'
+  pull_request:
+    paths:
+      - 'resources/plugin_uploader/**'
+      - '.github/workflows/test_plugin_uploader.yml'
+
+jobs:
+  unit-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: install deps
+        working-directory: resources/plugin_uploader
+        run: pip install -r requirements_test.txt
+
+      - name: run unit tests
+        working-directory: resources/plugin_uploader
+        run: pytest .
+
+  ruff-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - uses: chartboost/ruff-action@v1
+        with:
+          version: 0.4.10
+          src: './resources/plugin_uploader'
```

```suggestion
      - name: Lint with ruff
        run: |
          pip install ruff
          ruff check --output=github ./resources/plugin_uploader
```
i feel the github action `chartboost/ruff-action` does not bring a lot to the table and may be risky if they push a bad release. instead, i suggest using github's [example](https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-python#using-ruff-to-lint-code).